dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java

```diff
@@ -686,89 +728,116 @@ private int createCommand(CommandType commandType, long processDefineCode,
      * create complement command
      * close left and close right
      *
-     * @param start
-     * @param end
+     * @param scheduleTimeParam
      * @param runMode
      * @return
      */
-    protected int createComplementCommandList(Date start, Date end, RunMode runMode, Command command,
+    protected int createComplementCommandList(String scheduleTimeParam, RunMode runMode, Command command,
                                             Integer expectedParallelismNumber, ComplementDependentMode complementDependentMode) {
         int createCount = 0;
+        String startDate = null;
+        String endDate = null;
+        String dateList = null;
         int dependentProcessDefinitionCreateCount = 0;
-
         runMode = (runMode == null) ? RunMode.RUN_MODE_SERIAL : runMode;
         Map<String, String> cmdParam = JSONUtils.toMap(command.getCommandParam());
+        Map<String, String> scheduleParam = JSONUtils.toMap(scheduleTimeParam);
+        if(scheduleParam.containsKey(CMDPARAM_COMPLEMENT_DATA_SCHEDULE_DATE_LIST)){
+            dateList = scheduleParam.get(CMDPARAM_COMPLEMENT_DATA_SCHEDULE_DATE_LIST);
+        }
+        if(scheduleParam.containsKey(CMDPARAM_COMPLEMENT_DATA_START_DATE) && scheduleParam.containsKey(CMDPARAM_COMPLEMENT_DATA_END_DATE)){
+            startDate = scheduleParam.get(CMDPARAM_COMPLEMENT_DATA_START_DATE);
+            endDate = scheduleParam.get(CMDPARAM_COMPLEMENT_DATA_END_DATE);
+        }
         switch (runMode) {
             case RUN_MODE_SERIAL: {
-                if (start.after(end)) {
-                    logger.warn("The startDate {} is later than the endDate {}", start, end);
-                    break;
+                if(StringUtils.isNotEmpty(dateList) || dateList != null){
```

Comment:
I think ```StringUtils.isNotEmpty(null)``` would return false. So it's duplicated.

Reply:
ok